ckanext/protected_resources/plugin.py

```diff
@@ -142,8 +115,15 @@ def protected_resource_unlock(context, data_dict):
 
     resource = p.toolkit.get_action('resource_show')(context, {'id': data_dict['resource_id']})
 
-    if 'is_protected' in resource:
-        del resource['is_protected']
+    if not resource_has_protected_status(data_dict['resource_id']):
+        raise p.toolkit.ValidationError(['This resource is already unlocked'])
+    else:
+        log.error("WE're on the right track my dude")
```

Comment:
5/10
I feel like this log might need to go my dude :)

Reply:
How else will people know they are on the right track :thinking: 

_will fix_